numba_cuda/numba/cuda/device_init.py

```diff
@@ -87,8 +87,8 @@
     PTXSource,
 )
 
-from numba.cuda.misc.special import literal_unroll
-from numba.cuda.misc import literal
+# from numba.cuda.misc.special import literal_unroll
+# from numba.cuda.misc import literal
```

Comment:
I added this import so that the overload for `literal_unroll()` would get registered. Without it, the implementation won't be found. It doesn't necessarily need to be imported here (and if we want to avoid exposing it, we could do `del literal` afterwards) but it needs to get imported somehow when everything is initialized / imported.

Reply:
Something to get us off the ground at https://github.com/NVIDIA/numba-cuda/pull/523/commits/f210837cec34715408207ab0a8c5928febe8e234